code/__HELPERS/type2type.dm

```diff
@@ -99,6 +99,11 @@ GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,
 		else
 			return null
 
+///Returns a single dir rotated by x degrees.
+#define rotate_dir(dir, rotation) ( angle2dir( dir2angle(dir) + rotation ) )
+///Returns a single dir rotated by x degrees, adhering to the cardinal directions.
+#define rotate_dir_cardinal(dir, rotation) ( angle2dir_cardinal ( dir2angle(dir) + rotation ) )
```

Comment:
use turn() no? at least for rotate dir. also replace the decal code that does rotate_dir_cardinal with this macro (or one renamed to fit the turn() pattern)
also macros should typically be written with upper case